mvpa2/base/dataset.py

```diff
@@ -207,8 +211,17 @@ def __init__(self, samples, sa=None, fa=None, a=None):
 
         """
         # conversions
-        if isinstance(samples, list):
-            samples = np.array(samples)
+        if isinstance(samples, list) or isinstance(samples, tuple):
+            if np.all([is_datasetlike(e) for e in samples]):
```

Comment:
I guess faster one would be pure python 'all' on a generator so it could quit earlier:
```
if all(is_datasetlike(e) for e in samples)
```
?